src/app/space/create/deployments/apps/deployment-details.component.ts

```diff
@@ -241,4 +244,40 @@ export class DeploymentDetailsComponent {
     );
     return latch;
   }
+
+  private getTooltipContents(): any {
+    return {
+      contents: (d: any) => {
+        let tipRows: string = '';
+        let color = '#0088ce'; // pf-blue-400
+        let units: string = '';
+        if (d[0].name === 'CPU') {
+          units = this.cpuConfig.units;
+        } else if (d[0].name === 'Memory') {
+          units = this.memUnits;
+        }
+        for (let i = 0; i < d.length; i++) {
+          if (i === 0) {
+            tipRows += '<tr><th colspan="2">' + d[i].x.toLocaleString() + '</th></tr>';
```

Comment:
Can you add a comment explaining this special case for the first element? Is there an assumption on what `d` contains?

Reply:
From a bit of digging, `d` is the tooltip data [0]. I had modified this function from the default sparkline tooltip function in pf-ng [1], and fwiw, it looks like it was based on how c3 refers to it in their code as well [2]. 

For an example, I put a println into `getTooltipContents()` and got this for one of the CPU chart `d`: [{x: Wed Feb 21 2018 09:31:24 GMT-0500 (EST), value: 0, id: "CPU", index: 3, name: "CPU"}]. 

[0] http://c3js.org/reference.html#tooltip-contents
[1] https://github.com/patternfly/patternfly-ng/blob/master/src/app/chart/sparkline/sparkline.component.ts#L157
[2] https://github.com/c3js/c3/blob/master/c3.js#L9016